### PR TITLE
Fix multiple blur view issues

### PR DESCRIPTION
**Pull request by ja-ka**
Dimezis just released a fix with https://github.com/Dimezis/BlurView/issues/110 to address app crashes on Android when multiple blur views are in place. This PR includes the most recent version from https://github.com/Dimezis/BlurView into this project.

### CODE REVIEW

**Comment by flyingSAP** [comment]
Hi @Kureev can this be merged?

**Comment by zmtmaster** [comment]
Issue seems to be using two BlurView components, This PR uses the lasted update from https://github.com/Dimezis/BlurView which solves the issue,

Used this fix with patch-package it does work, so until this will be merged here's patch file if anyone need it.

```
diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
index 8177235..b401cb8 100644
--- a/node_modules/@react-native-community/blur/android/build.gradle
+++ b/node_modules/@react-native-community/blur/android/build.gradle
@@ -43,5 +43,5 @@ repositories {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'
-    implementation 'com.eightbitlab:blurview:1.6.3'
+    implementation 'com.eightbitlab:blurview:1.6.6'
 }
```

**Comment by tsdmrfth** [comment]
@zmtmaster Thanks for the patch. It works!

**Comment by nhannah** [comment]
Thanks for putting up the PR hopefully they merge it soon!

**Comment by lakhvirk** [comment]
Please merge it soon as this issue ruins my whole week to figure out where is the issue 😢 
I finished my project by testing app on iOS and it was working fine but when I tested on android, i saw my app getting freezes 🥶  and slow..
It took me so long to figure out this..
Please save someone’s life like me 👍 who might be struggling somewhere else 😃

**Comment by coding-st** [comment]
@zmtmaster thanks for the fix. Can you give a little more detailed description how to perform the fix with patch-package. I never used patch-package before & don't know what to do exactly.

Big thanks in advance.

**Comment by zmtmaster** [comment]
> @zmtmaster thanks for the fix. Can you give a little more detailed description how to perform the fix with patch-package. I never used patch-package before & don't know what to do exactly.
> 
> Big thanks in advance.

patch-package is a NPM module that allows you to change a node_module files (ie dependents, source files, type file etc') and by creating a diff file (patch file) it can keep the changes made for the whole team to be sync, it is great for use cases like this bug we are facing that we need to keep the module working but we can not wait for the PR to be merged. 

you can read more about it here https://www.npmjs.com/package/patch-package

**Comment by C-odes** [comment]
> Issue seems to be using two BlurView components, This PR uses the lasted update from https://github.com/Dimezis/BlurView which solves the issue,
> 
> See [Dimezis/BlurView#110 (comment)](https://github.com/Dimezis/BlurView/issues/110#issuecomment-786939860)
> 
> Used this fix with patch-package it does work, so until this will be merged here's patch file if anyone need it.
> 
> ```
> diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
> index 8177235..b401cb8 100644
> --- a/node_modules/@react-native-community/blur/android/build.gradle
> +++ b/node_modules/@react-native-community/blur/android/build.gradle
> @@ -43,5 +43,5 @@ repositories {
>  dependencies {
>      //noinspection GradleDynamicVersion
>      implementation 'com.facebook.react:react-native:+'
> -    implementation 'com.eightbitlab:blurview:1.6.3'
> +    implementation 'com.eightbitlab:blurview:1.6.6'
>  }
> ```

So I'm new to react native.. how do I add this as it's not merged yet?

**Comment by kevinvangelder** [comment]
@C-odes You put that snippet into a file at `patches/@react-native-community+blur+3.6.0.patch` and then, if you have patch-package correctly configured it will apply the patch after every yarn/npm install.

**Review by RodolfoGS — APPROVED** [review]

**Review by Gabrielfcs — APPROVED** [review]

**Comment by C-odes** [comment]
> @C-odes You put that snippet into a file at `patches/@react-native-community+blur+3.6.0.patch` and then, if you have patch-package correctly configured it will apply the patch after every yarn/npm install.

Applied the fix and patch-package. IT seemed to work but when I deleted node modules and ran yarn install. It didnt apply the patch... odd.  

IS there more than just applying the change, run patch-package to create the  "patches" folder with the changes in it?

**Comment by RodolfoGS** [comment]
> > @C-odes You put that snippet into a file at `patches/@react-native-community+blur+3.6.0.patch` and then, if you have patch-package correctly configured it will apply the patch after every yarn/npm install.
> 
> Applied the fix and patch-package. IT seemed to work but when I deleted node modules and ran yarn install. It didnt apply the patch... odd.
> 
> IS there more than just applying the change, run patch-package to create the "patches" folder with the changes in it?

You need to add `"postinstall": "patch-package"` to your `package.json` and install with `npm i patch-package`
Follow the setup: https://github.com/ds300/patch-package#set-up

**Comment by seancheung** [comment]
> Issue seems to be using two BlurView components, This PR uses the lasted update from https://github.com/Dimezis/BlurView which solves the issue,
> 
> See [Dimezis/BlurView#110 (comment)](https://github.com/Dimezis/BlurView/issues/110#issuecomment-786939860)
> 
> Used this fix with patch-package it does work, so until this will be merged here's patch file if anyone need it.
> 
> ```
> diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
> index 8177235..b401cb8 100644
> --- a/node_modules/@react-native-community/blur/android/build.gradle
> +++ b/node_modules/@react-native-community/blur/android/build.gradle
> @@ -43,5 +43,5 @@ repositories {
>  dependencies {
>      //noinspection GradleDynamicVersion
>      implementation 'com.facebook.react:react-native:+'
> -    implementation 'com.eightbitlab:blurview:1.6.3'
> +    implementation 'com.eightbitlab:blurview:1.6.6'
>  }
> ```

You don't need to do that. Just force the dependency version in your app/build.gradle:

```groovy
dependencies {
    implementation('com.eightbitlab:blurview:1.6.6') {
        force = true
    }
}
```

**Review by bouteillerAlan — APPROVED** [review]
work for me too

need this mr asap, this is annoying behavior while this package is useful ^^

**Review by Kureev — APPROVED** [review]

**Comment by Kureev** [comment]
Hi folks! Thanks for submitting a pull request, but this library is no longer maintained. Please, consider using a community fork: `@react-native-community/blur`

**Comment by sonicdoe** [comment]
Isn’t this @react-native-community/blur? [The `package.json` contains `@react-native-community/blur`](https://github.com/Kureev/react-native-blur/blob/v3.6.1/package.json#L2) and [react-native-community/react-native-blur](https://github.com/react-native-community/react-native-blur) redirects to this repository.

**Comment by C-odes** [comment]
> Isn’t this @react-native-community/blur? [The `package.json` contains `@react-native-community/blur`](https://github.com/Kureev/react-native-blur/blob/v3.6.1/package.json#L2) and [react-native-community/react-native-blur](https://github.com/react-native-community/react-native-blur) redirects to this repository.

Yes? It is @react-native-community/blur

**Comment by jaquinocode** [comment]
> Isn’t this @react-native-community/blur? [The `package.json` contains `@react-native-community/blur`](https://github.com/Kureev/react-native-blur/blob/v3.6.1/package.json#L2) and [react-native-community/react-native-blur](https://github.com/react-native-community/react-native-blur) redirects to this repository.

@Kureev Can you explain this? It's concerning. Why are you seemingly the owner of a community repo but you say you're not? Thanks. 🙂

For reference, this is what you said:
> Hi folks! Thanks for submitting a pull request, but this library is no longer maintained. Please, consider using a community fork: `@react-native-community/blur`

It makes 0 sense and a lot of people depend on this package.

**Comment by thiago-sants** [comment]
I'm having this problem too. Seems like it won't be fixed any time soon. And I didn't find other options. If anyone has an alternative, I would be really grateful.

**Comment by chenop** [comment]
> > Issue seems to be using two BlurView components, This PR uses the lasted update from https://github.com/Dimezis/BlurView which solves the issue,
> > See [Dimezis/BlurView#110 (comment)](https://github.com/Dimezis/BlurView/issues/110#issuecomment-786939860)
> > Used this fix with patch-package it does work, so until this will be merged here's patch file if anyone need it.
> > ```
> > diff --git a/node_modules/@react-native-community/blur/android/build.gradle b/node_modules/@react-native-community/blur/android/build.gradle
> > index 8177235..b401cb8 100644
> > --- a/node_modules/@react-native-community/blur/android/build.gradle
> > +++ b/node_modules/@react-native-community/blur/android/build.gradle
> > @@ -43,5 +43,5 @@ repositories {
> >  dependencies {
> >      //noinspection GradleDynamicVersion
> >      implementation 'com.facebook.react:react-native:+'
> > -    implementation 'com.eightbitlab:blurview:1.6.3'
> > +    implementation 'com.eightbitlab:blurview:1.6.6'
> >  }
> > ```
> 
> You don't need to do that. Just force the dependency version in your _android/app/build.gradle_:
> 
> ```groovy
> dependencies {
>     implementation('com.eightbitlab:blurview:1.6.6') {
>         force = true
>     }
> }
> ```

Adding 
```
implementation('com.eightbitlab:blurview:1.6.6') {
         force = true
}

```
Worked for me - can you please explain why is it fixing the issue?

**Comment by chenop** [comment]
Another way to resolve that is using the latest tagged version "3.6.1":
`"@react-native-community/blur": "https://github.com/Kureev/react-native-blur#3.6.1",`

Its include:

```
implementation('com.eightbitlab:blurview:1.6.6') {
         force = true
}
```

Its a shame it was not released...

**Comment by Lexical-Luke** [comment]
@chenop how exactly do I apply that? 

When I add that to my package.json and `npm install` I get the error "error: pathspec '3.6.1' did not match any file(s) known to git", I really need this fix as of course, this issue is still present.

**Comment by chenop** [comment]
@Lexical-Luke nothing special - like you wrote
- Add to package.json:
`"@react-native-community/blur": "https://github.com/Kureev/react-native-blur#3.6.1",`
- npm install

**Comment by alexanderblackh** [comment]
Anyone know how to fix this with iOS? Running 3.6.0, yarn says there is no 3.6.1.

**Comment by sayurimizuguchi** [comment]
Any timeline when we're going to release/publish 3.6.1?

**Comment by mjm918** [comment]
this issue still happening in 3.6.1

**Comment by premhowli-buyhatke** [comment]
I am using version 4.3.2 and still see the same issue. do I need to patch com.eightbitlab:blurview:1.6.6 or is there any new version available?

**Pull request merged**
